ffmpeg/decoder.c

```diff
@@ -211,24 +211,40 @@ int open_audio_decoder(input_params *params, struct input_ctx *ctx)
   return ret;
 }
 
+char* get_hw_decoder(int ff_codec_id)
+{
+    switch (ff_codec_id) {
+        case AV_CODEC_ID_H264:
+            return "h264_cuvid";
+        case AV_CODEC_ID_HEVC:
+            return "hevc_cuvid";
+        case AV_CODEC_ID_VP8:
+            return "vp8_cuvid";
+        case AV_CODEC_ID_VP9:
+            return "vp9_cuvid";
+        default:
+            return "";
+    }
+}
+
 int open_video_decoder(input_params *params, struct input_ctx *ctx)
 {
   int ret = 0;
   AVCodec *codec = NULL;
   AVFormatContext *ic = ctx->ic;
-
   // open video decoder
   ctx->vi = av_find_best_stream(ic, AVMEDIA_TYPE_VIDEO, -1, -1, &codec, 0);
   if (ctx->dv) ; // skip decoding video
   else if (ctx->vi < 0) {
     LPMS_WARN("No video stream found in input");
   } else {
     if (AV_HWDEVICE_TYPE_CUDA == params->hw_type) {
-      if (AV_CODEC_ID_H264 != codec->id) {
+      char* decoder_name = get_hw_decoder(codec->id);
+      if (!*decoder_name) {
         ret = lpms_ERR_INPUT_CODEC;
-        LPMS_ERR(open_decoder_err, "Non H264 codec detected in input");
+        LPMS_ERR(open_decoder_err, "Input codec does not support hardware acceleration");
```

Comment:
WDYT about adding a unit test for this i.e. passing in a non-supported codec for the decoder?

Reply:
Ah I see. Noticed [this comment](https://github.com/livepeer/lpms/blob/8f053cf92946f36cb2873a896b0db06713125142/ffmpeg/nvidia_test.go#L17) that should be updated then.